in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,192 @@
+
+The cost of the block in wei/mana is then the sum of the following:
+- the total L1 cost in wei divided by `target_mana_per_block`
+- `proving_cost_per_mana`
+
+## Getting the Base Fee in wei/mana
+
+After simulating a block, the mana used by each transaction is known.
+
+A proposer can compute a block's `total_mana_used` by summing the `mana_used` fields of all transactions in the block.
+
+This `total_mana_used` will be part of the L2 block header.
+
+The rollup contract will maintain a `excess_mana` variable for the proven and pending chains.
+
+The `excess_mana` is computed from the previous block as the difference between `total_mana_used` and `target_mana_per_block`.
+
+Compute the L2 congestion factor as the "fake exponential" of the ratio of `total_mana_used` plus the previous block's `excess_mana` to `target_mana_per_block`, and multiply by a constant factor of 1e9:
+
+```math
+\text{congestion factor} = \text{congestion factor multiplier} * \text{exp}\left(\frac{\text{total mana used} + \text{previous excess mana}}{\text{target mana per block}}\right)
+```
+
+The base fee in wei/mana is then:
+
+```math
+\text{base wei per mana} = \text{congestion factor} * \text{cost of block in wei/mana}
+```
+
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{overhead mana per tx} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base wei per mana} * \text{fee asset per wei}
+```
+
+## Distribution of the fee asset
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A fixed percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
+
+This quote is submitted by the prover in epoch `i+1` to prove epoch `i`, and is thus claimed/submitted to L1 by a proposer in epoch `i+1`.
+
+In order to prevent a prover bribing a proposer into accepting a quote with an extremely high fee, the maximum fee a prover can submit for an epoch proof quote is `maximum_epoch_proof_quote_fee`.
+
+## Clarifications and considerations
+
+### Why is `blobs_per_block` a constant?
+
+It makes computation easier. Otherwise block builder would need to know the number of blobs consumed by the transaction while it is building.
+
+### L1 Congestion
+
+If L1 gas spikes during an epoch, this will be reflected in the TST cost of a transaction through factoring in the higher `wei_per_l1_gas` and `wei_per_l1_blob_gas` into the TST cost.
```

Comment:
```suggestion
If L1 gas spikes during an epoch, this will be reflected in the `wei_per_l1_gas` and `wei_per_l1_blob_gas` oracles and make their way into the cost.
```